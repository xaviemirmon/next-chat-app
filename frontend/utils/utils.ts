export const isOnlyEmoji = (message: string) => {
  const trimmedMessage = message.trim();
  return trimmedMessage && /^[\p{Emoji}]+$/u.test(trimmedMessage);
};
