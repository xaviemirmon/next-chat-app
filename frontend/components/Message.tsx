import { isOnlyEmoji } from "@/utils/utils";
import styles from "./Message.module.css";
import { MessageType } from "@/types/types";

export const Message = ({
  message,
  user,
  grouped,
}: {
  message: MessageType;
  user: number | null;
  grouped: boolean;
}) => {
  return (
    <>
      <div className={getClassName(message, user, grouped)}>
        {message.content}
      </div>
    </>
  );
};

const getClassName = (
  message: MessageType,
  user: number | null,
  grouped: boolean,
) => {
  return `${styles.message} ${
    message.senderId === user ? styles.user : styles.sender
  } ${isOnlyEmoji(message.content) ? styles.emoji : ""} ${
    grouped ? styles.grouped : ""
  }`;
};
