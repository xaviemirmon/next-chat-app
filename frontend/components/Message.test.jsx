import { render } from "@testing-library/react";
import { Message } from "./Message";
import { isOnlyEmoji } from "@/utils/utils";

// Mocking the utility function
jest.mock("@/utils/utils", () => ({
  isOnlyEmoji: jest.fn(),
}));

describe("Message component", () => {
  const userId = 1;

  it("Matches snapshot for user message", () => {
    const message = {
      content: "Hello!",
      senderId: userId,
      createdAt: new Date().toISOString(),
    };

    const { asFragment } = render(
      <Message message={message} user={userId} grouped={false} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Matches snapshot for sender message", () => {
    const message = {
      content: "Hi there!",
      senderId: 2,
      createdAt: new Date().toISOString(),
    };

    const { asFragment } = render(
      <Message message={message} user={userId} grouped={false} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Matches snapshot for emoji-only message", () => {
    const message = {
      content: "ðŸ˜Š",
      senderId: 2,
      createdAt: new Date().toISOString(),
    };

    isOnlyEmoji.mockReturnValue(true);

    const { asFragment } = render(
      <Message message={message} user={userId} grouped={false} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Matches snapshot for grouped messages", () => {
    const message = {
      content: "This is a grouped message.",
      senderId: 2,
      createdAt: new Date().toISOString(),
    };

    const { asFragment } = render(
      <Message message={message} user={userId} grouped={true} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Matches snapshot for non-emoji message", () => {
    const message = {
      content: "How are you?",
      senderId: 2,
      createdAt: new Date().toISOString(),
    };

    isOnlyEmoji.mockReturnValue(false);

    const { asFragment } = render(
      <Message message={message} user={userId} grouped={false} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
