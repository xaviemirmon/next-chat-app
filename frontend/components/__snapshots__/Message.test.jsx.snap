// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component Matches snapshot for emoji-only message 1`] = `
<DocumentFragment>
  <div
    class="message sender emoji "
  >
    ðŸ˜Š
  </div>
</DocumentFragment>
`;

exports[`Message component Matches snapshot for grouped messages 1`] = `
<DocumentFragment>
  <div
    class="message sender emoji grouped"
  >
    This is a grouped message.
  </div>
</DocumentFragment>
`;

exports[`Message component Matches snapshot for non-emoji message 1`] = `
<DocumentFragment>
  <div
    class="message sender  "
  >
    How are you?
  </div>
</DocumentFragment>
`;

exports[`Message component Matches snapshot for sender message 1`] = `
<DocumentFragment>
  <div
    class="message sender  "
  >
    Hi there!
  </div>
</DocumentFragment>
`;

exports[`Message component Matches snapshot for user message 1`] = `
<DocumentFragment>
  <div
    class="message user  "
  >
    Hello!
  </div>
</DocumentFragment>
`;
